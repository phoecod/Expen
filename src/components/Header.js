import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux'
import {startLogout} from '../actions/auth';
export const Header = ({startLogout}) => {

	return (
		<header className="header">
			<h1>Expenso app</h1>
			<div className="nav-bar">
					<NavLink to="/dashboard" className="header-btn" activeClassName="is-active" exact={true}>Dashboard</NavLink>
					<NavLink to="/create" className="header-btn"  activeClassName="is-active">Create</NavLink>
					<button onClick={startLogout} className="header-btn" >Logout</button>
			</div>
		</header>
	)
}

const mapDispatchToProps = (dispatch) => ({
	startLogout: () => dispatch(startLogout())
})
export default connect(undefined, mapDispatchToProps)(Header);