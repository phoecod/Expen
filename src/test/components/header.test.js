import { shallow } from 'enzyme';
import React from 'react';
import Header from '../../components/header';

test('should render Header correctly', () => {
	const wrapper = shallow(<Header />);
	expect(wrapper).toMatchSnapshot();
});