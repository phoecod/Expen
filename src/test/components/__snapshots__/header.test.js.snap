// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <h1>
    Expenso app
  </h1>
  <div
    className="nav-bar"
  >
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      className="header-btn"
      to="/dashboard"
    >
      Dashboard
    </NavLink>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      className="header-btn"
      to="/create"
    >
      Create
    </NavLink>
    <button
      className="header-btn"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</header>
`;
